modules/react-native-wallet/android/src/main/java/com/expensify/wallet/GooglePushProvisioningModule.java

```diff
@@ -0,0 +1,233 @@
+            }
+          }
+        }
+      })
+      .addOnFailureListener(new OnFailureListener() {
+        @Override
+        public void onFailure(@NonNull Exception e) {
+          Log.i(TAG, "onFailure (getActiveWalletID) - " + e.getMessage());
+          promise.reject("GET_ACTIVE_WALLET_ID_FAILURE", e);
+        }
+      })
+      .addOnCanceledListener(new OnCanceledListener() {
+        @Override
+        public void onCanceled() {
+          Log.i(TAG, "onCanceled (getActiveWalletID) - ");
+          promise.reject("GET_ACTIVE_WALLET_ID_CANCELED", "Active wallet ID retrieval canceled");
+        }
+      });
+  }
+
+  @ReactMethod
+  public void getStableHardwareId(Promise promise) {
+    tapAndPayClient.getStableHardwareId()
+      .addOnCompleteListener(new OnCompleteListener<String>() {
+        @Override
+        public void onComplete(@NonNull Task<String> task) {
+          if (task.isSuccessful()) {
+            promise.resolve(task.getResult());
+          } else {
+            promise.reject("GET_STABLE_HARDWARE_ID_ERROR", task.getException());
+          }
+        }
+      })
+      .addOnFailureListener(new OnFailureListener() {
+        @Override
+        public void onFailure(@NonNull Exception e) {
+          promise.reject("GET_STABLE_HARDWARE_ID_FAILURE", e);
+        }
+      })
+      .addOnCanceledListener(new OnCanceledListener() {
+        @Override
+        public void onCanceled() {
+          promise.reject("GET_STABLE_HARDWARE_ID_CANCELED", "Stable hardware ID retrieval canceled");
+        }
+      });
+  }
+
+  @ReactMethod
+  public void getEnvironment(Promise promise) {
+    tapAndPayClient.getEnvironment()
+      .addOnCompleteListener(new OnCompleteListener<String>() {
+        @Override
+        public void onComplete(@NonNull Task<String> task) {
+          if (task.isSuccessful()) {
+            promise.resolve(task.getResult());
+          } else {
+            promise.reject("GET_ENVIRONMENT_ERROR",  task.getException());
+          }
+        }
+      })
+      .addOnFailureListener(new OnFailureListener() {
+        @Override
+        public void onFailure(@NonNull Exception e) {
+          promise.reject("GET_ENVIRONMENT_FAILURE", e);
+        }
+      })
+      .addOnCanceledListener(new OnCanceledListener() {
+        @Override
+        public void onCanceled() {
+          promise.reject("GET_ENVIRONMENT_CANCELED", "Environment retrieval canceled");
+        }
+      });
+  }
+
+  @ReactMethod
+  public void pushProvision(String opc, String tsp, String clientName, String lastDigits, String addressJson, Promise promise) {
+    try {
+      this.promise = promise;
+      JSONObject address = new JSONObject(addressJson);
+      int cardNetwork = tsp.equals("VISA") ? TapAndPay.CARD_NETWORK_VISA : TapAndPay.CARD_NETWORK_MASTERCARD;
+      int tokenServiceProvider = tsp.equals("VISA") ? TapAndPay.TOKEN_PROVIDER_VISA : TapAndPay.TOKEN_PROVIDER_MASTERCARD;
```

Comment:
@brunovjk Please make `VISA` a string constant.

Also refer to the previous comment regarding token service providers.